src/test/java/org/broadinstitute/barclay/help/BashTestDoclet.java

```diff
@@ -0,0 +1,143 @@
+     *         required
+     *
+     *         Each of those keys maps to a List&lt;SimpleMap&gt; representing each property.
+     *         These property maps each contain the following keys:
+     *
+     *             kind
+     *             name
+     *             summary
+     *             fulltext
+     *             otherArgumentRequired
+     *             synonyms
+     *             exclusiveOf
+     *             type
+     *             options
+     *             attributes
+     *             required
+     *             minRecValue
+     *             maxRecValue
+     *             minValue
+     *             maxValue
+     *             defaultValue
+     *
+     */
+    @Override
+    protected void emitOutputFromTemplates (
+            final List<Map<String, String>> groupMaps,
+            final List<Map<String, String>> featureMaps)
+    {
+        try {
+            /* ------------------------------------------------------------------- */
+            /* You should do this ONLY ONCE in the whole application life-cycle:   */
+            final Configuration cfg = new Configuration();
+            cfg.setDirectoryForTemplateLoading(settingsDir); // where the template files come from
+            cfg.setObjectWrapper(new DefaultObjectWrapper());
+
+            // Create a root map for all the work units so we can access all the info we need:
+            Map<String, Object> propertiesMap = new HashMap<>();
+            workUnits.stream().forEach( workUnit -> propertiesMap.put(workUnit.getName(), workUnit.getRootMap()) );
+
+            // Add everything into a nice package that we can iterate over
+            // while exposing the command line program names as keys:
+            Map<String, Object> rootMap = new HashMap<>();
+            rootMap.put("tools", propertiesMap);
+
+            // Get or create a template
+            final Template template = cfg.getTemplate(getIndexTemplateName());
+
+            // Create the output file
+            final File indexFile = new File(getDestinationDir()
+                    + File.separator
+                    + getIndexBaseFileName()
+                    + "."
+                    + getIndexFileExtension()
+            );
+
+            // Run the template and merge in the data
+            try (final FileOutputStream fileOutStream = new FileOutputStream(indexFile);
+                 final OutputStreamWriter outWriter = new OutputStreamWriter(fileOutStream)) {
+                template.process(rootMap, outWriter);
+            } catch (TemplateException e) {
+                throw new DocException("Freemarker Template Exception during documentation index creation", e);
+            }
+
+        } catch (FileNotFoundException e) {
+            throw new RuntimeException("FileNotFoundException processing javadoc template", e);
+        } catch (IOException e) {
+            throw new RuntimeException("IOException processing javadoc template", e);
+        }
+    }
```

Comment:
That's a good point.  It's pretty trivial to move the logic into processIndexTemplate - I'll update it.